File: python-client-sdk/src/sdk/models/shared/codat_datacontracts_datasets_billcreditnotepagedresponselinksmodel.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import codat_datacontracts_datasets_billcreditnotepagedresponsehrefmodel as shared_codat_datacontracts_datasets_billcreditnotepagedresponsehrefmodel


@dataclass_json
@dataclasses.dataclass
class CodatDataContractsDatasetsBillCreditNotePagedResponseLinksModel:
    current: Optional[shared_codat_datacontracts_datasets_billcreditnotepagedresponsehrefmodel.CodatDataContractsDatasetsBillCreditNotePagedResponseHrefModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('current') }})
    next: Optional[shared_codat_datacontracts_datasets_billcreditnotepagedresponsehrefmodel.CodatDataContractsDatasetsBillCreditNotePagedResponseHrefModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('next') }})
    previous: Optional[shared_codat_datacontracts_datasets_billcreditnotepagedresponsehrefmodel.CodatDataContractsDatasetsBillCreditNotePagedResponseHrefModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('previous') }})
    self: Optional[shared_codat_datacontracts_datasets_billcreditnotepagedresponsehrefmodel.CodatDataContractsDatasetsBillCreditNotePagedResponseHrefModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('self') }})
    

File: typescript-client-sdk/src/sdk/models/shared/codatdatacontractsdatasetsbillcreditnotepagedresponselinksmodel.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { CodatDataContractsDatasetsBillCreditNotePagedResponseHrefModel } from "./codatdatacontractsdatasetsbillcreditnotepagedresponsehrefmodel";
import { Expose, Type } from "class-transformer";


export class CodatDataContractsDatasetsBillCreditNotePagedResponseLinksModel extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "current" })
  @Type(() => CodatDataContractsDatasetsBillCreditNotePagedResponseHrefModel)
  current?: CodatDataContractsDatasetsBillCreditNotePagedResponseHrefModel;

  @SpeakeasyMetadata()
  @Expose({ name: "next" })
  @Type(() => CodatDataContractsDatasetsBillCreditNotePagedResponseHrefModel)
  next?: CodatDataContractsDatasetsBillCreditNotePagedResponseHrefModel;

  @SpeakeasyMetadata()
  @Expose({ name: "previous" })
  @Type(() => CodatDataContractsDatasetsBillCreditNotePagedResponseHrefModel)
  previous?: CodatDataContractsDatasetsBillCreditNotePagedResponseHrefModel;

  @SpeakeasyMetadata()
  @Expose({ name: "self" })
  @Type(() => CodatDataContractsDatasetsBillCreditNotePagedResponseHrefModel)
  self?: CodatDataContractsDatasetsBillCreditNotePagedResponseHrefModel;
}